sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -1618,12 +1618,18 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
   }
 
   test("SPARK-9511: error with table starting with number") {
-    val df = sqlContext.sparkContext.parallelize(1 to 10).map(i => (i, i.toString))
-      .toDF("num", "str")
-    df.registerTempTable("1one")
-
-    checkAnswer(sql("select count(num) from 1one"), Row(10))
+    withTempTable("1one") {
+      sqlContext.sparkContext.parallelize(1 to 10).map(i => (i, i.toString))
+        .toDF("num", "str")
+        .registerTempTable("1one")
+      checkAnswer(sql("select count(num) from 1one"), Row(10))
+    }
+  }
 
-    sqlContext.dropTempTable("1one")
+  test("SPARK-9955: correct error message for aggregate") {
+    val e = intercept[AnalysisException] {
+      sql("select key, max(value) from testData where bad_column > 1 group by key")
```

Comment:
Or construct the plan by hand?  Or we could maybe start a dataframe error
suite.
On Aug 14, 2015 8:26 PM, "Wenchen Fan" <notifications@github.com> wrote:

> In sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala
> <https://github.com/apache/spark/pull/8203#discussion_r37132757>:
>
> >
> > -    sqlContext.dropTempTable("1one")
> > +  test("SPARK-9955: correct error message for aggregate") {
> > +    val e = intercept[AnalysisException] {
> > +      sql("select key, max(value) from testData where bad_column > 1 group by key")
>
> The UnresolvedAlias is only introduced when parsing SQL or DataFrame, so
> if we move this test into AnalysisErrorSuite, we need to add the
> UnresolvedAliases manually, is that OK?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/8203/files#r37132757>.
>